Makefile

```diff
@@ -158,12 +158,52 @@ docker-multi-stage:
 	@echo ">> building docker image 'thanos' with Dockerfile.multi-stage"
 	@docker build -f Dockerfile.multi-stage -t "thanos" .
 
-# docker-push pushes docker image build under `thanos` to "$(DOCKER_IMAGE_REPO):$(DOCKER_IMAGE_TAG)"
+# docker-cross builds docker images with multiple arches.
+# We use quay.io/prometheus/busybox:latest as base image which is quay.io/prometheus/busybox:uclibc indeed.
+# Details refer to https://quay.io/repository/prometheus/busybox.
+# Since in CircleCI env, there is no amd64 arch machine, we can not directly pull it with arm64 arch.
+# So build images for amd64 and arm64 arch here.
+.PHONY: docker-cross
+docker-cross: build
+	@echo ">> building docker image for different arch"
+	@git clone https://github.com/prometheus/busybox.git
+	@pushd busybox && \
+		./build.sh "quay.io/prometheus/busybox-linux-amd64" "" "" uclibc && \
+		./build.sh "quay.io/prometheus/busybox-linux-arm64" "arm64v8/" "" uclibc && \
+		popd
+	@docker run --privileged linuxkit/binfmt:v0.7
+	@docker build -f Dockerfile.cross --build-arg ARCH=amd64 -t thanos-linux-amd64 .
+	@docker build -f Dockerfile.cross --build-arg ARCH=arm64 -t thanos-linux-arm64 .
+
+# docker-test test docker images with multiple arches
+.PHONY: docker-test
+docker-test:
+	@echo ">> testing image"
+	@docker run thanos-linux-amd64 --help
+	@docker run thanos-linux-arm64 --help
+
+# docker-test tag docker images with multiple arches
+.PHONY: docker-tag
+docker-tag:
+	@echo ">> taging image"
+	@docker tag thanos-linux-amd64 "${DOCKER_IMAGE_REPO}-linux-amd64:$(DOCKER_IMAGE_TAG)"
+	@docker tag thanos-linux-arm64 "${DOCKER_IMAGE_REPO}-linux-arm64:$(DOCKER_IMAGE_TAG)"
+
+# docker-manifest push docker manifest to support multiple arches
+.PHONY: docker-manifest
+docker-manifest:
+	@echo ">> creating and pushing manifest"
+	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest create -a "$(DOCKER_IMAGE_REPO):$(DOCKER_IMAGE_TAG)" \
+							"$(DOCKER_IMAGE_REPO)-linux-amd64:$(DOCKER_IMAGE_TAG)" \
+							"$(DOCKER_IMAGE_REPO)-linux-arm64:$(DOCKER_IMAGE_TAG)"
+	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest push "$(DOCKER_IMAGE_REPO):$(DOCKER_IMAGE_TAG)"
+
+# docker-push pushes docker image with multiple arches
 .PHONY: docker-push
 docker-push:
 	@echo ">> pushing image"
-	@docker tag "thanos" "$(DOCKER_IMAGE_REPO):$(DOCKER_IMAGE_TAG)"
-	@docker push "$(DOCKER_IMAGE_REPO):$(DOCKER_IMAGE_TAG)"
+	@docker push "$(DOCKER_IMAGE_REPO)-linux-amd64:$(DOCKER_IMAGE_TAG)"
```

Comment:
Why push twice?

Reply:
dockerhub and quay.io, notice that there are 2 images.